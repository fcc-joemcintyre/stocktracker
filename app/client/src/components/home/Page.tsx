import { Range } from '../range/Range';
import { StockChart } from '../stockChart/StockChart';
import { StockEntry } from '../stockEntry/StockEntry';
import { Status } from '../status/Status';
import { StockCard } from '../stock/StockCard';
import { Stock } from '../app/types.js';

type Props = {
  data: Stock[],
  months: number,
  retrieving: string[],
  errors: string[],
  onRangeChanged: (months: number) => void,
  onAddStock: (symbol: string) => void,
  onRemoveStock: (symbol: string) => void,
};

export const Page = ({
  data, months, retrieving, errors, onRangeChanged, onAddStock, onRemoveStock,
}: Props) => {
  const colors = ['blue', 'green', 'lightblue', 'lightgreen', 'purple', 'orange', 'lightpurple', 'steelblue'];
  const stockData = data.map ((item, index) => ({ ...item, color: colors[index] }));

  const stocks = data.map ((item, index) => (
    <StockCard
      key={item.symbol}
      symbol={item.symbol}
      name={item.name}
      color={colors[index]}
      onRemoveStock={onRemoveStock}
    />
  ));

  return (
    <>
      <div className='titlebar'>
        <h1 className='title'>StockTracker</h1>
      </div>
      <div style={{ marginTop: '4px', marginBottom: '4px' }}>
        <Range
          months={months}
          onRangeChanged={onRangeChanged}
        />
      </div>
      <StockChart
        width={900}
        height={500}
        months={months}
        stocks={stockData}
      />
      <div style={{ display: 'flex' }}>
        { stocks }
      </div>
      { stocks.length < 8 && (
        <StockEntry onAddStock={onAddStock} />
      )}
      <Status requests={retrieving} errors={errors} />
    </>
  );
};
