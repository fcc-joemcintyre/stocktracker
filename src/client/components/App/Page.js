import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Titlebar } from '../Titlebar';
import { Range } from '../Range';
import { StockChart } from '../StockChart';
import { Stock } from '../Stock';
import { StockEntry } from '../StockEntry';
import { Status } from '../Status';

export const Page = ({ data, months, retrieving, errors, chartKey, onRangeChanged, onAddStock, onRemoveStock }) => {
  const colors = ['blue', 'green', 'lightblue', 'lightgreen', 'purple', 'orange', 'lightpurple', 'steelblue'];
  const stockData = data.map ((item, index) => Object.assign ({}, item, { color: colors[index] }));
  const stocks = data.map ((item, index) => (
    <Stock
      key={item.symbol}
      symbol={item.symbol}
      name={item.name}
      color={colors[index]}
      onRemoveStock={onRemoveStock}
    />
  ));

  // if not all tracking slots (8) are taken, show entry component
  const stockEntry = (stocks.length < 8) ? <StockEntry onAddStock={onAddStock} /> : null;

  return (
    <Fragment>
      <Titlebar />
      <Range months={months} onRangeChanged={onRangeChanged} />
      <StockChart
        key={chartKey}
        className='stockChart'
        width={900}
        height={500}
        months={months}
        stocks={stockData}
      />
      <div className='stockArea'>
        {stocks}
        {stockEntry}
      </div>
      <Status requests={retrieving} errors={errors} />
    </Fragment>
  );
};

Page.propTypes = {
  data: PropTypes.arrayOf (PropTypes.shape ({})).isRequired,
  months: PropTypes.number.isRequired,
  retrieving: PropTypes.arrayOf (PropTypes.string).isRequired,
  errors: PropTypes.arrayOf (PropTypes.string).isRequired,
  chartKey: PropTypes.number.isRequired,
  onRangeChanged: PropTypes.func.isRequired,
  onAddStock: PropTypes.func.isRequired,
  onRemoveStock: PropTypes.func.isRequired,
};
