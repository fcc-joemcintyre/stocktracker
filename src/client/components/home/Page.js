import React from 'react';
import PropTypes from 'prop-types';
import { Range } from '../range/Range';
import { StockChart } from '../stockChart/StockChart';
import { StockEntry } from '../stockEntry/StockEntry';
import { Status } from '../status/Status';
import { Stock } from '../stock/Stock';

export const Page = ({ data, months, retrieving, errors, onRangeChanged, onAddStock, onRemoveStock }) => {
  const colors = ['blue', 'green', 'lightblue', 'lightgreen', 'purple', 'orange', 'lightpurple', 'steelblue'];
  const stockData = data.map ((item, index) => ({ ...item, color: colors[index] }));

  const stocks = data.map ((item, index) => (
    <Stock
      key={item.symbol}
      symbol={item.symbol}
      name={item.name}
      color={colors[index]}
      onRemoveStock={onRemoveStock}
    />
  ));

  return (
    <>
      <div className='titlebar'>
        <h1 className='title'>StockTracker</h1>
      </div>
      <div style={{ marginTop: '4px', marginBottom: '4px' }}>
        <Range
          months={months}
          onRangeChanged={onRangeChanged}
        />
      </div>
      <StockChart
        className='stockChart'
        width={900}
        height={500}
        months={months}
        stocks={stockData}
      />
      <div style={{ display: 'flex' }}>
        { stocks }
      </div>
      { stocks.length < 8 && (
        <StockEntry onAddStock={onAddStock} />
      )}
      <Status requests={retrieving} errors={errors} />
    </>
  );
};

Page.propTypes = {
  data: PropTypes.arrayOf (PropTypes.shape ({})).isRequired,
  months: PropTypes.number.isRequired,
  retrieving: PropTypes.arrayOf (PropTypes.string).isRequired,
  errors: PropTypes.arrayOf (PropTypes.string).isRequired,
  onRangeChanged: PropTypes.func.isRequired,
  onAddStock: PropTypes.func.isRequired,
  onRemoveStock: PropTypes.func.isRequired,
};
