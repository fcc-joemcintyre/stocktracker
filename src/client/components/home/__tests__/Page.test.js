/* eslint-env jest */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Page } from '../Page';

test ('Page starting state', () => {
  render (
    <Page
      data={[]}
      months={3}
      retrieving={[]}
      errors={[]}
      chartKey={0}
      onRangeChanged={jest.fn ()}
      onAddStock={jest.fn ()}
      onRemoveStock={jest.fn ()}
    />
  );
  expect (screen.getByRole ('heading')).toBeDefined ();
  expect (screen.getByRole ('heading')).toHaveTextContent ('StockTracker');
});
