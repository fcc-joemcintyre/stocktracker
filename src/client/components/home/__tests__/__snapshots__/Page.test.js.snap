// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page starting state 1`] = `
Array [
  <div
    className="titlebar"
  >
    <div
      className="title"
    >
      StockTracker
    </div>
  </div>,
  <div
    style={
      Object {
        "marginBottom": "4px",
        "marginTop": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "border": "1px solid #7f7f7f",
            "borderRadius": "8px",
            "overflow": "hidden",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderLeft": null,
              "color": "#ffffff",
              "display": "inline-block",
              "padding": "4px 12px",
            }
          }
        >
          1m
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "teal",
              "borderLeft": "1px solid #ffffff",
              "color": "#ffffff",
              "display": "inline-block",
              "padding": "4px 12px",
            }
          }
        >
          3m
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderLeft": "1px solid #ffffff",
              "color": "#ffffff",
              "display": "inline-block",
              "padding": "4px 12px",
            }
          }
        >
          6m
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderLeft": "1px solid #ffffff",
              "color": "#ffffff",
              "display": "inline-block",
              "padding": "4px 12px",
            }
          }
        >
          1y
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderLeft": "1px solid #ffffff",
              "color": "#ffffff",
              "display": "inline-block",
              "padding": "4px 12px",
            }
          }
        >
          3y
        </div>
      </div>
      <div
        flex="1 1 0"
      />
    </div>
  </div>,
  <svg
    height={500}
    width={900}
  />,
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  />,
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="symbolText"
    >
      Symbol:
    </label>
     
    <input
      id="symbolText"
      onChange={[Function]}
      style={
        Object {
          "width": "60px",
        }
      }
      type="text"
      value=""
    />
     
    <button
      disabled={true}
      type="submit"
    >
      Track
    </button>
  </form>,
  <div>
    <div>
      Pending requests: 
      none
    </div>
    <div>
      Errors: 
      none
    </div>
  </div>,
]
`;
